app/js/components/identifiers/addIdentifier.jsx

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import PropTypes from 'react-proptypes';
+import apiCall from '../../utilities/apiHelper';
+
+class NewIdentifier extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            newState: false,
+            addIdentifiers: {},
+            editErrors: {
+                uuid: '',
+                error: '',
+            },
+            addErrors: {
+                uuid: '',
+                error: '',
+            },
+        };
+        this.callNew = this.callNew.bind(this);
+        this.callCreate = this.callCreate.bind(this);
+        this.handleCreateChange = this.handleCreateChange.bind(this);
+        this.callCancelPromise = this.callCancelPromise.bind(this);
+    }
+
+    callNew() {
+        this.setState({ newState: true });
+    }
+
+    handleCreateChange(e) {
+        const label = e.target.name;
+        const value = e.target.value;
+        const { addIdentifiers } = this.state;
+
+        if (label === 'preferred') {
+            addIdentifiers.preferred = e.target.checked;
+        } else {
+            addIdentifiers[label] = value;
+        }
+        this.setState(Object.assign({}, this.state.addIdentifiers, {
```

Comment:
Could you use functional `setState` here? You can read about it [here](https://medium.com/@shopsifter/using-a-function-in-setstate-instead-of-an-object-1f5cfd6e55d1) or [here](https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b)

Reply:
Done.